### DISCUSSION TITLE
Expose host servce through docker nginx-proxy

### DISCUSSION THREAD

**Discussion by tomasz-grobelny** [nginx-proxy/nginx-proxy#2143] [Q&A]
Hi all,

I have nginx-proxy running on docker and exposed on port 80 and 443. Next to that, on the same machine, I have another http service exposed on custom port (eg. 1234). Is there any way I could make the nginx-proxy treat this service as a backend? Is it a valid use case from nginx-proxy point of view? Or maybe any workaround exists to make it happen? I can imagine that I would need a trivial docker container that would expose port 80 and forward all requests back to the host - any thoughts how to implement it the simplest way? Eg. any preexiting docker containers implementing this?

**Comment by rhansen** [comment]
> on the same machine, I have another http service exposed on custom port (eg. 1234). Is there any way I could make the nginx-proxy treat this service as a backend?

Yes, assuming the other service is in a Docker container and you want it served over the existing nginx ports (80 and 443).  Just set the `VIRTUAL_HOST` environment variable as described in the documentation.

If the other service is not in a Docker container, or if you want `nginx-proxy` to serve it on port 1234, then no, that is not directly supported yet.  You can [inject some custom nginx directives](https://github.com/nginx-proxy/nginx-proxy#custom-nginx-configuration) to accomplish what you want, however.
